src/lib/plugin_apis/lvm.api

```diff
@@ -330,6 +353,74 @@ GType bd_lvm_lvdata_get_type () {
     return type;
 }
 
+#define BD_LVM_TYPE_VDODATA (bd_lvm_lvdata_get_type ())
+GType bd_lvm_vdodata_get_type();
+
+/**
+ * BDLVMVDOPooldata:
+ * @operationg_mode: operationg mode of the VDO pool (e.g. 'normal')
+ * @compression_state: state of the compression
+ * @index_state: state of the VDO index
+ * @saving_percent: percentage of physical blocks saved
+ * @deduplication: whether deduplication is enabled
+ * @compression: whether compression is enabled
+ */
+
+typedef struct BDLVMVDOPooldata {
+    BDLVMVDOOperatingMode operating_mode;
+    BDLVMVDOCompressionState compression_state;
+    BDLVMVDOIndexState index_state;
+    guint64 used_size;
+    guint64 saving_percent;
```

Comment:
Not sure about the negative values. I chose 64bit because other similar values like `data_percent` in lvinfo are also 64bit.

Reply:
I have changed the type to gint32. LVM internally uses `dm_percent_t` for it, which is int32_t and it can be `-1`(or `DM_PERCENT_INVALID`) when the pool is not operating in the normal mode.